azcertcache_test.go

```diff
@@ -2,146 +2,156 @@ package azcertcache_test
 
 import (
 	"context"
+	"fmt"
 	"io/ioutil"
-	"reflect"
+	"strings"
 	"testing"
 
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/stretchr/testify/assert"
 	"golang.org/x/crypto/acme/autocert"
 
 	"github.com/goenning/azcertcache"
 )
 
-func expectNotNil(t *testing.T, v interface{}) {
-	if v == nil {
-		t.Errorf("should not be nil")
-	}
-}
+const accountName = "devstoreaccount1"
+const accountKey = "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw=="
 
-func expectNil(t *testing.T, v interface{}) {
-	if v != nil {
-		t.Errorf("should be nil, but was %v", v)
-	}
-}
+// Bad account key encoded in base64 that would fail the authentication check
+const badAccountKey = "YmFkY3JlZGVudGlhbA==" // base64: "badcredential"
 
-func expectEquals(t *testing.T, v interface{}, expected interface{}) {
-	if !reflect.DeepEqual(v, expected) {
-		t.Errorf("should be %v, but was %v", expected, v)
-	}
-}
+func newCache(t *testing.T, containerPrefix string, accountKey string) (*azcertcache.Cache, error) {
+	// Append test name after the container prefix to make them different.
```

Comment:
Can this still have namespace collisions if multiple people run the same test at the same time? Maybe also add a random string to the container name?

Reply:
Good point, but this is outside of expected use case. Doubt we are going to have people running tests at the same time. Besides, I don't even know if the storage account in this test is still running. 

I'll add comments to tell people to use their own Azure storage account name and keys at the top of this file.

Reply:
that is not a real azure account, the tests target a local container (described on docker-compose.yml)